Missing packages installed by addon's install blueprint

I have an addon that installs NPM packages using the addon's install blueprint.

e.g.

`/blueprints/<addon-name>/index.js`

```js
/*jshint node:true*/
module.exports = {
  description: '',
  normalizeEntityName: function() {},

  afterInstall: function() {
    return this.addPackageToProject('ceibo');
  }
};
```

I'm using `ceibo` here but it could be any other NPM package.

The problem is that this package is not present when I create more than one application using `ember-cli-addon-tests`.

The package is defined in the `package.json` of the first app created by `ember-cli-addon-tests` but is not present in the `package.json` of the following apps created by `ember-cli-addon-tests`.

e.g.

```js
it('first test', function() {
  app = new AddonTestApp();
  return app.create('first-app').then(function() {
    assert.ok(fs.readFileSync(app.path('package.json').indexOf('ceibo') > -1)); // runs OK
  });
});

it('second test', function() {
  app = new AddonTestApp();
  return app.create('second-app').then(function() {
    assert.ok(fs.readFileSync(app.path('package.json').indexOf('ceibo') > -1)); // FAILS
  });
});
```

I've created a project with the minimum code to reproduce the issue [here](https://github.com/san650/add-package-to-project-issue).

You can see it fails [here](https://travis-ci.org/san650/add-package-to-project-issue/jobs/185308270).

@simonihmig Thanks for your response. I've run the tests using the DEBUG flag, here's the output

<details>
<summary>Output</summary>

```
$ DEBUG=ember-cli-addon-tests npm test

> add-package-to-project-issue@1.0.0 test /Projects/add-package-to-project-issue
> mocha --recursive test/

  01-test
  ember-cli-addon-tests created tmp; path=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg +0ms
  ember-cli-addon-tests no precooked node_modules +1ms
  ember-cli-addon-tests chdir; path=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg +1ms
  ember-cli-addon-tests no node_modules or bower_components +1ms
  ember-cli-addon-tests chdir; path=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine +0ms
  ember-cli-addon-tests running command=/Projects/add-package-to-project-issue/node_modules/ember-cli/bin/ember; args=new,--disable-analytics,--watcher=node,--skip-git,dummy; cwd=/private/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine +4ms
  ember-cli-addon-tests chdir; path=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine/dummy +2m
  ember-cli-addon-tests installing ember-data emberjs/data#master +0ms
  ember-cli-addon-tests installing ember-disable-prototype-extensions +3ms
  ember-cli-addon-tests installed ember-disable-prototype-extension +8s
  ember-cli-addon-tests installed ember-data emberjs/data#master +13s
  ember-cli-addon-tests found addon package.json; path=/Projects/add-package-to-project-issue/package.json +3ms
  ember-cli-addon-tests found addon package.json; path=/Projects/add-package-to-project-issue/package.json +2ms
  ember-cli-addon-tests symlinking add-package-to-project-issue +0ms
  ember-cli-addon-tests found addon package.json; path=/Projects/add-package-to-project-issue/package.json +5ms
  ember-cli-addon-tests installing add-package-to-project-issue as addon for application +0ms
  ember-cli-addon-tests installed ember canary +719ms
  ember-cli-addon-tests found addon package.json; path=/Projects/add-package-to-project-issue/package.json +3ms
  ember-cli-addon-tests running add-package-to-project-issue generator +2ms
  ember-cli-addon-tests running command=/Projects/add-package-to-project-issue/node_modules/ember-cli/bin/ember; args=generate,add-package-to-project-issue; cwd=/private/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine/dummy +0ms
  ember-cli-addon-tests DEPRECATION: Node v5.10.1 is no longer supported by Ember CLI. We recommend that you use the most-recent "Active LTS" version of Node.js.
 +830ms
  ember-cli-addon-tests WARNING: Node v5.10.1 is not tested against Ember CLI on your platform. We recommend that you use the most-recent "Active LTS" version of Node.js.
 +1ms
  ember-cli-addon-tests installing add-package-to-project-issue
 +47ms
  ember-cli-addon-tests   install addon ember-cli-mirage
 +8ms
  ember-cli-addon-tests Installed packages for tooling via npm.
 +16s
  ember-cli-addon-tests installing ember-cli-mirage
 +166ms
  ember-cli-addon-tests   create /mirage/config.js
 +13ms
  ember-cli-addon-tests   create /mirage/scenarios/default.js
 +0ms
  ember-cli-addon-tests   create /mirage/serializers/application.js
 +0ms
  ember-cli-addon-tests Installed addon package.
 +16ms
  ember-cli-addon-tests copying pristine app; from=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine/dummy; to=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/under-test/dummy +38ms
  ember-cli-addon-tests copying complete +32ms
  ember-cli-addon-tests chdir; path=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/under-test/dummy +0ms
  ember-cli-addon-tests chdir; path=/Projects/add-package-to-project-issue +1ms
  ember-cli-addon-tests copying fixtures; from=/Projects/add-package-to-project-issue/test/fixtures/dummy; to=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/under-test/dummy +0ms
    ✓ foo test
    ✓ bar test

  02-test
  ember-cli-addon-tests chdir; path=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg +14ms
  ember-cli-addon-tests existing node_modules and bower_components +0ms
  ember-cli-addon-tests chdir; path=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine +0ms
  ember-cli-addon-tests running command=/Projects/add-package-to-project-issue/node_modules/ember-cli/bin/ember; args=new,--skip-bower,--skip-npm,--disable-analytics,--watcher=node,--skip-git,second; cwd=/private/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine +1ms
  ember-cli-addon-tests chdir; path=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine/second +530ms
  ember-cli-addon-tests installing ember-data emberjs/data#master +1ms
  ember-cli-addon-tests found addon package.json; path=/Projects/add-package-to-project-issue/package.json +2ms
  ember-cli-addon-tests installing add-package-to-project-issue as addon for application +0ms
  ember-cli-addon-tests found addon package.json; path=/Projects/add-package-to-project-issue/package.json +2ms
  ember-cli-addon-tests running add-package-to-project-issue generator +2ms
  ember-cli-addon-tests running command=/Projects/add-package-to-project-issue/node_modules/ember-cli/bin/ember; args=generate,add-package-to-project-issue; cwd=/private/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine/second +0ms
  ember-cli-addon-tests
Missing bower packages:
Package: ember
  * Specified: canary
  * Installed: (not installed)

Package: ember-cli-shims
  * Specified: 0.1.3
  * Installed: (not installed)

Run `bower install` to install missing dependencies.

 +503ms
  ember-cli-addon-tests copying pristine app; from=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/pristine/second; to=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/under-test/second +6ms
  ember-cli-addon-tests copying complete +30ms
  ember-cli-addon-tests chdir; path=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/under-test/second +0ms
  ember-cli-addon-tests chdir; path=/Projects/add-package-to-project-issue +0ms
  ember-cli-addon-tests copying fixtures; from=/Projects/add-package-to-project-issue/test/fixtures/second; to=/var/folders/j4/g0k_9zns07v0ptp8l0dw117h0000gn/T/d-1161120-51636-1ss3ojg/under-test/second +1ms
    1) foo test

    2) bar test



  2 passing (3m)
  2 failing

  1) 02-test foo test:

      AssertionError: "ember-cli-mirage" not found in package.json
      + expected - actual

      -false
      +true

      at assertPackage (test/02-test.js:12:12)
      at Context.<anonymous> (test/02-test.js:22:5)

  2) 02-test bar test:

      AssertionError: "ember-cli-mirage" not found in package.json
      + expected - actual

      -false
      +true

      at assertPackage (test/02-test.js:12:12)
      at Context.<anonymous> (test/02-test.js:26:5)



npm ERR! Test failed.  See above for more details.
```
</details>

Do you see anything strange on the output?

--- a/lib/utilities/pristine.js
+++ b/lib/utilities/pristine.js
@@ -122,9 +122,6 @@ function installPristineApp(appName, options) {
     });
   }
 
-  // at this point we have all deps available, so we can run ember-cli
-  promise = promise.then(runAddonGenerator);
-
   if (!hasNodeModules) {
     promise = promise.then(movePristineNodeModules(appName));
   }
@@ -133,7 +130,9 @@ function installPristineApp(appName, options) {
     promise = promise.then(movePristineBowerComponents(appName));
   }
 
-  return promise.then(linkDependencies(appName));
+  return promise.then(linkDependencies(appName))
+    // at this point we have all deps available, so we can run ember-cli
+    .then(runAddonGenerator);
 }
 
 function hasPristineNodeModules() {

